ertviz/controllers/response_correlation_controller.py

```diff
@@ -0,0 +1,273 @@
+            parameter_df = ensemble.parameters_df(parameters)
+            response = responses[0]
+            response_df = ensemble.responses[response].data_df()
+            parameter_df[response] = response_df.loc[corr_index]
+            corrdf = parameter_df.corr(method=correlation_metric)
+            corrdf = corrdf.reindex(corrdf[response].abs().sort_values().index)
+            data[ens_key] = corrdf[response].drop(response, axis=0).dropna()
+            data[ens_key].index.name = response
+            colors[ens_key] = color["color"]
+        if bool(data):
+            parent.correlation_plot = BarChartPlotModel(data, colors)
+            return parent.correlation_plot.repr
+
+    @app.callback(
+        Output(
+            {"id": parent.uuid("response-overview"), "type": parent.uuid("graph")},
+            "figure",
+        ),
+        [
+            Input(parent.uuid("correlation-store-xindex"), "modified_timestamp"),
+            Input(parent.uuid("parameter-selection-store-resp"), "modified_timestamp"),
+            Input(parent.uuid("ensemble-selection-store"), "modified_timestamp"),
+        ],
+        [
+            State(parent.uuid("parameter-selection-store-resp"), "data"),
+            State(parent.uuid("ensemble-selection-store"), "data"),
+            State(parent.uuid("correlation-store-xindex"), "data"),
+        ],
+    )
+    def update_response_overview_plot(_, __, ___, responses, ensembles, corr_xindex):
+        if not ensembles or not responses:
+            raise PreventUpdate
+
+        _plots = []
+        for ensemble_id, data in ensembles.items():
+            ensemble = load_ensemble(parent, ensemble_id)
+            # take the first selected response
+            response = ensemble.responses[responses[0]]
+            x_axis = response.axis
+            if str(x_axis[0]).isnumeric():
+                style = deepcopy(assets.ERTSTYLE["response-plot"]["response-index"])
+            else:
+                style = deepcopy(assets.ERTSTYLE["response-plot"]["response"])
+            data_df = response.data_df()
+
+            style["marker"]["color"] = data["color"]
+            _plots += [
+                PlotModel(
+                    x_axis=x_axis,
+                    y_axis=data_df[x_val],
+                    text="Mean",
+                    name=f"Mean {repr(ensemble)}",
+                    **style,
+                )
+                for x_val in data_df
+            ]
+
+            for obs in response.observations:
+                _plots += _get_observation_plots(obs.data_df(), x_axis)
+        fig = go.Figure()
+        for plot in _plots:
+            fig.add_trace(plot.repr)
+
+        _layout = assets.ERTSTYLE["figure"]["layout"].copy()
+        _layout.update(dict(showlegend=False))
+        fig.update_layout(_layout)
+        fig.add_shape(
+            type="line",
+            x0=x_axis[corr_xindex],
+            y0=0,
+            x1=x_axis[corr_xindex],
```

Comment:
are you using the last assigned x_axis from the `for` loop starting on line 93 by purpose at this point? ( might make sense, just curious if it is by purpose and will be correct so just wanted to double check with you )

Reply:
Yes, this one shows the selections line. It is sort-of expected that all ensembles have the same `x_axis` for the same response; therefore the last x_axis is as good as the first. If there are no ensembles (ie. no response) there is this 
```python
if not ensembles or not responses:
            raise PreventUpdate
```
 in the beginning. Can be updated if you have a suggestion...